### ISSUE TITLE
Missing "with_counts" for "Get Guild"

### ISSUE THREAD

**Issue by LoadingBG** [IGJoshua/discljord#66]
In [the docs for "Get Guild"](https://discord.com/developers/docs/resources/guild#get-guild) it says that you can provide an optional `with_counts` field so that the response contains `approximate_member_count` and `approximate_presence_count`. Looking at `discljord.messaging/get-guild!`, I don't see how I can pass this optional field. Is it possible to pass `with_counts` as an argument somehow and if so, how?

**Comment by IGJoshua** [comment]
You are correct that it's missing. This is absolutely trivial to add, so I'll try to get to it tonight, but in the meantime you could evaluate the following snippet to expose a `:with-counts` keyword argument (although it won't have completion available from in your editor) and it will replace the existing get-guild.

```clojure
(let [old-ns *ns*]
  (require 'discljord.messaging.impl)
  (in-ns 'discljord.messaging.impl)
  (defdispatch :get-guild
    [guild-id] [with-counts] _ :get _ body
    (str "/guilds/" guild-id)
    {:body (json/write-str {:with_counts with-counts})}
    (json-body body))
  (in-ns old-ns))
```

**Comment by LoadingBG** [comment]
Leiningen doesn't let me do that. I get all kinds of `Unable to resolve symbol` errors. I tried to give everything a namespace prefix but I got `var #'discljord.messaging.impl/json-body is not public`.

**Comment by IGJoshua** [comment]
I've just put out v1.2.1 which includes a fix for this!

**Comment by LoadingBG** [comment]
Using `discljord.messaging/get-guild!` now gives me:
```console
Jan 20, 2021 10:34:18 AM clojure.tools.logging$eval10137$fn__10140 invoke
INFO: Starting messaging process
Jan 20, 2021 10:34:18 AM clojure.tools.logging$eval10137$fn__10140 invoke
INFO: Connecting shard 0
2021-01-20 10:34:18.101:INFO::async-dispatch-3: Logging initialized @5987ms to org.eclipse.jetty.util.log.StdErrLog
Jan 20, 2021 10:34:21 AM clojure.tools.logging$eval10137$fn__10140 invoke
SEVERE: Exception in dispatch-http
java.lang.Exception: JSON error (unexpected character): <
        at clojure.data.json$_read.invokeStatic(json.clj:230)
        at clojure.data.json$_read.invoke(json.clj:181)
        at clojure.data.json$read.invokeStatic(json.clj:276)
        at clojure.data.json$read.doInvoke(json.clj:232)
        at clojure.lang.RestFn.invoke(RestFn.java:410)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:660)
        at clojure.data.json$read_str.invokeStatic(json.clj:282)
        at clojure.data.json$read_str.doInvoke(json.clj:278)
        at clojure.lang.RestFn.invoke(RestFn.java:410)
        at discljord.messaging.impl$json_body.invokeStatic(impl.clj:80)
        at discljord.messaging.impl$json_body.invoke(impl.clj:78)
        at discljord.messaging.impl$eval11148$fn__11150.invoke(impl.clj:329)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at discljord.messaging.impl$make_request_BANG_$make_request__12350.invoke(impl.clj:796)
        at discljord.messaging.impl$make_request_BANG_.invokeStatic(impl.clj:801)
        at discljord.messaging.impl$make_request_BANG_.invoke(impl.clj:775)
        at discljord.messaging.impl$step_agent$fn__12374.invoke(impl.clj:846)
        at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2033)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.Agent$Action.doRun(Agent.java:114)
        at clojure.lang.Agent$Action.run(Agent.java:163)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```
It doesn't matter whether I use `:with-counts` or not.

**Comment by JohnnyJayJay** [comment]
Yeah, The issue is that @IGJoshua didn't add this as a URL parameter but to the body of the request. He's probably not available right now, but I will make a PR for this so it's resolved asap.

**Comment by LoadingBG** [comment]
Thank you.

**Comment by IGJoshua** [comment]
That's what I get for making a hotfix tired. i'll have that merged soon and another patch version out

**Comment by LoadingBG** [comment]
What is the patched version?

**Comment by IGJoshua** [comment]
It should be fixed in 1.2.2 which was released on clojars today

**Comment by SamTheone1** [comment]
Thank
